### PR TITLE
Separate tasklists and calendars

### PR DESCRIPTION
**Pull request by marbetschar**
Separates Task and Calendars in the list view of the CalDAV dialog:

- ~~dropped `sort_func` because it did not add any value on my end (sources seem to be sorted correctly already)~~
- Moved discovering logic into `find_sources_supporting` which is capable to distinguish between tasklists and calendars which allows us to set the later needed extension values directly
- Added section headers for "Calendars" and "Task Lists" in the newly added `header_func`
- improved `sort_func` so calendars and tasklists are separated

### CODE REVIEW

File: src/Dialogs/CaldavDialog.vala [hunk 1]
```diff
@@ -288,11 +288,35 @@ public class OnlineAccounts.CaldavDialog : Hdy.Window {
...
+        } else {
+            var before_source_row = (SourceRow) before;
+            var before_is_calendar = before_source_row.source.has_extension (E.SOURCE_EXTENSION_CALENDAR);
+            var before_is_tasklist = before_source_row.source.has_extension (E.SOURCE_EXTENSION_TASK_LIST);
+
+            if (before_is_calendar && is_tasklist) {
+                header_label = new Granite.HeaderLabel (_("Task Lists"));
+            } else if (before_is_tasklist && is_calendar) {
+                header_label = new Granite.HeaderLabel (_("Calendars"));
+            }
```
Comment (danrabbit) [inline src/Dialogs/CaldavDialog.vala:314]: Because you are adding calendars to the liststore and then adding tasklists, this seems like an impossible case

**Review by danrabbit — CHANGES_REQUESTED** [review]
I'm no longer getting sources sorted in alphabetical order here, which is not a huge deal, but it could be fixed by using `insert_sorted` as in https://github.com/elementary/code/blob/master/src/Services/GitManager.vala#L47

One inline comment about the header func

**Comment by marbetschar** [comment]
@danrabbit re-added `sort_func` and improved it a bit. Looks good now on my end. Can you please re-review?

**Review by danrabbit — APPROVED** [review]
LFG

**Pull request merged**